src/components/ChatWoot.ts

```diff
@@ -7,6 +7,41 @@ export const ChatwootWidget: React.FC = () => {
   const chatwootEnabled = import.meta.env.VITE_FEATURE_CHATWOOT === 'true'
   useEffect(() => {
     if (!chatwootEnabled) return // Add Chatwoot Settings
+    const BASE_URL = getConfig().VITE_CHATWOOT_URL
+    let allowedOrigin = ''
+    try {
+      allowedOrigin = new URL(BASE_URL).origin
+    } catch {
+      allowedOrigin = ''
+    }
```

Comment:
nitpick
```suggestion
    const allowedOrigin = (() => {
      try {
        return new URL(BASE_URL).origin
      } catch {
        return ''
      }
    })()
```

Reply:
Fair, functional is the way!

https://github.com/shapeshift/web/pull/10374/commits/302e53fceb4158bec7a0e2664420bf6ff3396c09